lang/compiler.go

```diff
@@ -337,3 +340,127 @@ func (p *AstProgram) Start(ctx context.Context, alloc *memory.Allocator) (flux.Q
 
 	return p.Program.Start(ctx, alloc)
 }
+
+func (p *AstProgram) eval(ctx context.Context, deps dependencies.Interface) ([]interpreter.SideEffect, values.Scope, error) {
+	s, _ := opentracing.StartSpanFromContext(ctx, "eval")
+
+	sideEffects, scope, err := flux.EvalAST(ctx, deps, p.Ast, flux.SetNowOption(p.Now))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	s.Finish()
+	s, _ = opentracing.StartSpanFromContext(ctx, "compile")
+	defer s.Finish()
+
+	nowOpt, ok := scope.Lookup(flux.NowOption)
+	if !ok {
+		return nil, nil, fmt.Errorf("%q option not set", flux.NowOption)
+	}
+
+	nowTime, err := nowOpt.Function().Call(ctx, deps, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	p.Now = nowTime.Time().Time()
+	return sideEffects, scope, nil
+}
+
+func (p *AstProgram) updateOpts(scope values.Scope) error {
+	planOpts, err := getPlanOptions(scope)
+	if err != nil {
+		return err
+	}
+	p.opts.planOptions.logical = append(p.opts.planOptions.logical, planOpts.logical...)
+	p.opts.planOptions.physical = append(p.opts.planOptions.physical, planOpts.physical...)
+	return nil
+}
+
+func getPlanOptions(scope values.Scope) (planOptions, error) {
+	planOpt, ok := scope.Lookup("planner")
+	if !ok {
+		// No option specified.
+		return planOptions{}, nil
+	}
+	if planOpt.Type().Nature() != semantic.Object {
+		return planOptions{}, fmt.Errorf("option 'planner' must be an object, got %s", planOpt.Type().Nature().String())
+	}
+	var err error
+	var wrongKeysErr = fmt.Errorf("the only available field for option 'planner' is 'disable'")
+	if planOpt.Object().Len() > 3 {
+		return planOptions{}, wrongKeysErr
+	}
+	planOpt.Object().Range(func(name string, v values.Value) {
+		// TODO(affo): we could add 'enable' and 'only'.
+		//  In order to do this we should add the possibility to enable rules by name in the planner, first.
+		if name != "disable" {
+			err = wrongKeysErr
+		}
+	})
+	if err != nil {
+		return planOptions{}, err
+	}
+
+	la, pa, err := getRuleArraysForKey("disable", planOpt)
```

Comment:
I see your point. In general we follow the convention of using short names for local variables:

- https://www.reddit.com/r/golang/comments/8wxwgv/why_does_go_encourage_short_variable_names/
- https://www.quora.com/Why-does-Golang-promote-short-and-kind-of-meaningless-names-for-variables

The idea is that if a variable lives for a short period of time and lives only in the context of a function, then it is more readable to use a few letters for it (as for `ses` above).

In this case, for example, `la` and `pa` come back from a function that contains the name `Array` in it, and they are "cast" to `Array()` later. To me, it seems an excess to use `logicalArray` because the lifespan of the variable is veri limited, and its nature is trivially understandable.

This discussion is purely subjective and you might argue that `logicalArray` and `physicalArray` are better. I would say that their names don't metter at all because they are used only to build slices 2 lines below 😅 .

We use this style throughout the Flux codebase, and after getting use to it, I actually love it.